client/src/app/components/transaction-history/transaction-history.component.ts

```diff
@@ -0,0 +1,34 @@
+import { Component, OnInit, Input, OnChanges } from '@angular/core';
+import { Transaction } from 'src/app/interfaces/transaction';
+
+@Component({
+  selector: 'app-transaction-history',
+  templateUrl: './transaction-history.component.html',
+  styleUrls: ['./transaction-history.component.scss']
+})
+export class TransactionHistoryComponent implements OnInit, OnChanges {
+  @Input() userTransactions: Transaction[];
+  transactionHistory: Transaction[];
+  constructor() { }
+
+  ngOnInit(): void {
+
+  }
+  ngOnChanges(): void {
+    if (this.userTransactions) { this.transactionHistory = this.userTransactions; }
```

Comment:
Can you explain why you need an ngOnChanges?
Secondly, if the whole purpose of the ```transactionHistory``` variable is to be a copy of the ```userTransactions``` input, why don't you just use the ```userTransactionsInput```?